File: deployment/deploy.py
#!/usr/bin/env python3

from web3_interface import Web3Interface
import json
from datetime import datetime
from contract import Contract
from log import log_address
from tx_args import tx_args

web3 = Web3Interface().w3
if web3.net.version == "1":
  sender_account = "0x54d9249C776C56520A62faeCB87A00E105E8c9Dc"
else:
  sender_account = web3.eth.accounts[0]
compiled_path = "./build/"
contract_name = "Crowdsale"
contract = Contract()
print("\nDeploying contract...")
tx_hash = contract.deploy(compiled_path, contract_name, tx_args(sender_account, gas=4500000),)
print("\nDeployment transaction hash: ", tx_hash.hex(),
      "\nCrowdsale address: ", contract.contract.address)
log_address(contract.contract)

File: deployment/deploy.js
//Load modules
var fs = require("fs");
var solc = require('solc');
var Web3 = require('web3');
var net = require('net');

//Instantiate variables
var gasEstimated;
var gasPriceEstimated;
var contractInstance;
var contractAddress;
var mainAddress;

//Address that will handle tokens trapped in the contract
var tokenRetrieverAccount = "0x0F048ff7dE76B83fDC14912246AC4da5FA755cFE";

//Connect to the private node
var web3 = new Web3(new Web3.providers.IpcProvider('/home/coinfabrik/Programming/blockchain/node/geth.ipc', net));

//Import params into config variable
var config = require("../config.js")(web3, "privateTestnet");

//Read abi and bytecode from files
var abi = require("./Crowdsale.abi.js");
var bytecode = require("./Crowdsale.bin.js");

function getAccount(){
  web3.eth.getAccounts(function(error, result) {
    if(error != null) {
      console.log("Couldn't get accounts");
    }
    else {
      mainAddress = result[0];
    }
  })
}

getAccount();

//Create contract instance
var crowdsaleContract = new web3.eth.Contract(abi, {
   from: mainAddress,
   gasPrice: 2000000000000000000,
   gas: 4700000,
   data: bytecode
 });

//Find estimated gas price for operation
web3.eth.getGasPrice().then((averageGasPrice) => {
  console.log("Average gas price: " + averageGasPrice);
  gasPriceEstimated = averageGasPrice;
}).catch(console.error);

function deployContract(){

  console.log("The address of Mathias account is: " + mainAddress);
  //crowdsaleContract.deploy({ arguments: [config.multisig_owners[0], config.startTime, config.endTime, account, config.tranches] }).estimateGas(function(err, gas){
  //  console.log("Estimated gas: " + gas);
  //  gasEstimated = gas;
  //});

  crowdsaleContract.deploy({ arguments: [config.multisig_owners[0], config.startTime, config.endTime, tokenRetrieverAccount, config.tranches] }).send({
      from: mainAddress,
      gasPrice: gasPriceEstimated, 
      gas: 47000000000
  }).then(function(instance) {
      console.log(instance.options.address);
      contractInstance = instance;
      contractAddress = instance.options.address;
  });
}

setTimeout(deployContract, 20);